jflex/src/test/java/jflex/state/StateSetQuickcheck.java

```diff
@@ -181,10 +195,26 @@ public void addStateAdds(StateSet set, @InRange(minInt = 0, maxInt = 34) int e)
   }
 
   @Property
-  public void addStateDoesNotRemove(StateSet set, @InRange(minInt = 0, maxInt = 34) int e) {
+  public void addStateDoesNotRemove(
+      StateSet set, @Also("2147483647") @InRange(minInt = 0, maxInt = 34) int e) {
     StateSet setPre = new StateSet(set);
     set.addState(e);
     assertThat(set.contains(setPre)).isTrue();
+    assertThat(set.hasElement(e)).isTrue();
+
+    // add an out of range value to increase coverage of contains
+
+    // offset to StateSetGen.maxRange + 1
+    int offset =
+        1001; // note this effected by InRange, so this needs to be adjusted based on annotations
+    // on set, default is set
+
+    //  if no overflow then offset + e, else overflow so use MAX_VALUE
+    int newValue = (Integer.MAX_VALUE - offset) >= e ? offset + e : Integer.MAX_VALUE;
+    set.addState(newValue);
+    assertThat(set.contains(setPre)).isTrue();
+    assertThat(set.hasElement(newValue)).isTrue();
+    assertThat(setPre.contains(set)).isFalse();
```

Comment:
This condition is not necessarily true: if the new elements that were added happened to be in the set before, the set will not have changed.

Reply:
@lsf37  This line I think will take care of this:
assumeThat(set.hasElement(newValue), equalTo(false));

Reply:
Indeed it will. Thanks, that resolves the remaining issue.